dhis-2/dhis-services/dhis-service-administration/src/test/java/org/hisp/dhis/resourcetable/table/AbstractNameUniquenessAwareResourceTableTest.java

```diff
@@ -0,0 +1,72 @@
+package org.hisp.dhis.resourcetable.table;
+
+import static org.apache.commons.lang.StringUtils.countMatches;
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.hisp.dhis.category.Category;
+import org.hisp.dhis.category.CategoryOptionGroupSet;
+import org.hisp.dhis.common.CodeGenerator;
+import org.hisp.dhis.common.DataDimensionType;
+import org.hisp.dhis.indicator.IndicatorGroupSet;
+import org.junit.Test;
+
+public class AbstractNameUniquenessAwareResourceTableTest
+{
+    @Test
```

Comment:
I am wondering if it is better to have this as a stateful util class, e.g. `UniqueNameVerifier` or similar. Using the class hierarchy for very specific stuff might be limiting us/causing issues going forward.